### PR TITLE
feat(report): added CycloneDX report

### PR DESCRIPTION
**Pull request by rafaela-soares**
**Proposed Changes**
- Now, KICS can export results in CycloneDX XML format

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -36,7 +36,11 @@ kics.config
 results.*
 gl-sast-results.json
 sonarqube-results.json
+<<<<<<< HEAD
+cyclonedx-results.xml
+=======
 junit-results.xml
+>>>>>>> master
```
Comment (felipe-avelar) [inline .gitignore]:
```suggestion
cyclonedx-results.xml
junit-results.xml
```

File: pkg/report/model/cyclonedx.go [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+func getPurl(filePath, version string) string {
+	return fmt.Sprintf("pkg:generic/%s@%s", filePath, version)
+}
+
+func getDescription(query *model.QueryResult) string {
+	queryDescription := query.Description
+
+	fmt.Println(query.CISDescriptionTextFormatted)
+	if query.CISDescriptionTextFormatted != "" {
+		queryDescription = fmt.Sprintf("[CIS]: %s", query.CISDescriptionTextFormatted)
```
Comment (felipe-avelar) [inline pkg/report/model/cyclonedx.go:155]: I think we could remove this tag...

File: pkg/report/model/cyclonedx.go [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+		XMLNSV:       "http://cyclonedx.org/schema/ext/vulnerability/1.0",
+		SerialNumber: "urn:uuid:" + uuid.New().String(),
+		Version:      1,
+		Metadata:     &metadata,
+	}
+}
+
+// BuildCycloneDxReport builds the CycloneDX report
+func BuildCycloneDxReport(summary *model.Summary) *CycloneDxReport {
+	var c Component
```
Comment (felipe-avelar) [inline pkg/report/model/cyclonedx.go:222]: Based on the comment in line 258, I think this could be removed

File: pkg/report/model/cyclonedx.go [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+	var version, sha, purl, filePath string
+
+	bom := InitCycloneDxReport()
+	files := getAllFiles(summary)
+
+	for i := range files {
+		filePath = strings.Replace(files[i].FileName, "\\", "/", -1)
+		sha = generateSha256(filePath)
+
+		index := 12
```
Comment (felipe-avelar) [inline pkg/report/model/cyclonedx.go:233]: I think we could define this as a constant outside the loop with some name like `minShaLength`

File: pkg/report/model/cyclonedx.go [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+			Hashes: []Hash{
+				{
+					Alg:     "SHA-256",
+					Content: sha,
+				},
+			},
+			Vulnerabilities: vuln,
+		}
+
+		bom.Components.Components = append(bom.Components.Components, c)
```
Comment (felipe-avelar) [inline pkg/report/model/cyclonedx.go:258]: I think we could just define this here, like following suggestion:
```suggestion
		component := Component{
			Type:    "file",
			BomRef:  purl,
			Name:    filePath,
			Version: version,
			Purl:    purl,
			Hashes: []Hash{
				{
					Alg:     "SHA-256",
					Content: sha,
				},
			},
			Vulnerabilities: vuln,
		}

		bom.Components.Components = append(bom.Components.Components, component )
```

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Please, check comments. Also, there is a security issue about opening a file, please check this too...

**Review by felipe-avelar — DISMISSED** [review]
LGTM

File: pkg/report/model/cyclonedx.go [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+							Recommendation: fmt.Sprintf("In line %d, a result was found. '%s', but '%s'",
+								file.Line, file.KeyActualValue, file.KeyExpectedValue),
+						},
+					},
+				}
+				vulns = append(vulns, vuln)
+			}
+		}
+	}
+	return vulns
```
Comment (joaoReigota1) [inline pkg/report/model/cyclonedx.go:191]: I think this function could be refactored for readability, can you place from lines 163 to 189 in a separate function?

**Review by joaoReigota1 — APPROVED** [review]
LGTM

**Pull request merged**